addons/web_editor/static/src/js/wysiwyg/widgets/image_crop_widget.js

```diff
@@ -1,174 +1,228 @@
-                    ratio = this.aspectRatioList[i][2];
-                    break;
-                }
-            }
-            this.$cropperImage.cropper({
-                viewMode: 2,
-                dragMode: 'move',
-                autoCropArea: 1.0,
-                aspectRatio: ratio,
-                data: _.pick(data, 'x', 'y', 'width', 'height', 'rotate', 'scaleX', 'scaleY')
+    async start() {
+        if (!this.isCroppable) {
+            this.displayNotification({
+              type: 'warning',
+              title: _t("This image is an external image"),
+              message: _t("This type of image is not supported for cropping.<br/>If you want to crop it, please first download it from the original source and upload it in Odoo."),
             });
+            return this.destroy();
         }
-        return this._super.apply(this, arguments);
+        const _super = this._super.bind(this);
+        const cropperWrapper = this.$('.o_we_cropper_wrapper')[0];
+
+        // Replacing the src with the original's so that the layout is correct.
+        this.media.setAttribute('src', this.originalSrc);
+        await new Promise(resolve => this.media.addEventListener('load', resolve, {once: true}));
+        this.$cropperImage = this.$('.o_we_cropper_img');
+        const cropperImage = this.$cropperImage[0];
+        [cropperImage.style.width, cropperImage.style.height] = [this.$media.width() + 'px', this.$media.height() + 'px'];
+
+        // Overlaying the cropper image over the real image
+        $(cropperWrapper).offset(this.$media.offset());
+        // The cropper takes the size of the container, including padding
+        // The image doesn't fill its padding, so we use margin on the container
+        // to replicate the effect on the cropper preview.
+        [cropperWrapper.style.marginLeft, cropperWrapper.style.marginTop] = [this.$media.css('padding-left'), this.$media.css('padding-top')];
+
+        cropperImage.setAttribute('src', this.originalSrc);
+        await new Promise(resolve => cropperImage.addEventListener('load', resolve, {once: true}));
+        this.$cropperImage.cropper({
+            viewMode: 2,
+            dragMode: 'move',
+            autoCropArea: 1.0,
+            aspectRatio: this.aspectRatios[this.aspectRatio].value,
+            data: _.mapObject(_.pick(this.media.dataset, ...cropperDataFields), value => parseFloat(value)),
+            // Can't use 0 because it's falsy and the lib will then use its defaults (200x100)
+            minContainerWidth: 1,
+            minContainerHeight: 1,
+        });
+        core.bus.trigger('deactivate_snippet');
+
+        this._onOutsideClick = this._onOutsideClick.bind(this);
+        // We use capture so that the handler is called before other editor handlers
+        // like save, such that we can restore the src before a save.
+        this.document.addEventListener('mousedown', this._onOutsideClick, {capture: true});
```

Comment:
Not only that, we also need to destroy it before the editor gets a hold of the event because otherwise, if you click the crop button twice, it will try to create a new cropper with the wrong data as we replaced the image's src with that of the original.